pkg/test/framework/components/istio/operator.go

```diff
@@ -215,6 +212,25 @@ func deploy(ctx resource.Context, env *kube.Environment, cfg Config) (Instance,
 			return nil, err
 		}
 	}
+	// If control plane cluster isn't the config cluster we are under external control plane mode which uses remote KUBECONFIG
+	// we follow below install sequences:
+	// install istiod Remote + base in remote cluster first.
+	// install istiod in external control plane cluster
+	// get istiod loadbalancer address(discoveryAddress)
+	// create istiod service and endpoints in remote/config cluster
+	for _, cluster := range env.KubeClusters {
+		if env.IsControlPlaneCluster(cluster) && !env.IsConfigCluster(cluster) {
+			err = deployWithRemoteKubeConfig(iopFile, remoteIopFile, env, cfg, i)
+			return i, err
+		}
+	}
+
+	// For multicluster, create and push the CA certs to all clusters to establish a shared root of trust.
+	if env.IsMulticluster() {
```

Comment:
Let's move this up above where we start installing control planes.

Reply:
thinking we need to put this refactor into another separate PR, this change in this PR is big enough

Reply:
A lot of my concern is complexity introduced by this PR. I'd rather have your change simplify rather than make it more complicated.

Reply:
This means a big refactor for current `deploy` logic .  I do not want to mix this up with the `external control plane` test as it is hard to debug and fix when there are issues.

Reply:
external control plane is put into `deployWithRemoteKubeConfig` specially and not introduce complexity into existing  code logic